src/curlMultiHandler.hpp

```diff
@@ -111,11 +115,17 @@ class cURLMultiHandler final : public ICURLHandler
                 if (multiHandleMessages && (multiHandleMessages->msg == CURLMSG_DONE))
                 {
                     auto errorCode = multiHandleMessages->data.result;
+
+                    // Check for cURL-level errors (network, DNS, timeout, etc.)
                     if (errorCode != CURLE_OK)
                     {
-                        throw Curl::CurlException("cURLMultiHandler::execute() failed: " +
-                                                      std::string(curl_easy_strerror(errorCode)),
-                                                  errorCode);
+                        throw Curl::CurlException(curl_easy_strerror(errorCode), -1); 
```

Comment:
Why don't we use `NOT_USED`?
```suggestion
                        throw Curl::CurlException(curl_easy_strerror(errorCode), NOT_USED); 
```

Reply:
Solved!